src/components/Avatar.js

```diff
@@ -17,43 +21,45 @@ const propTypes = {
     containerStyles: stylePropTypes,
 
     /** Set the size of Avatar */
-    size: PropTypes.oneOf(['default', 'small']),
+    size: PropTypes.oneOf(_.values(CONST.AVATAR_SIZE)),
 
-    /** Whether this avatar is for a chat room */
-    isChatRoom: PropTypes.bool,
-
-    /** Whether this avatar is for an archived default room */
-    isArchivedRoom: PropTypes.bool,
+    /** The fill color for the icon. Can be hex, rgb, rgba, or valid react-native named color such as 'red' or 'blue' */
+    fill: PropTypes.string,
 };
 
 const defaultProps = {
-    source: '',
+    source: null,
     imageStyles: [],
     containerStyles: [],
-    size: 'default',
-    isChatRoom: false,
-    isArchivedRoom: false,
+    size: CONST.AVATAR_SIZE.DEFAULT,
+    fill: themeColors.icon,
 };
 
 class Avatar extends PureComponent {
     render() {
-        if (!this.props.source && !this.props.isChatRoom) {
+        if (!this.props.source) {
             return null;
         }
 
         const imageStyle = [
-            this.props.size === 'small' ? styles.avatarSmall : styles.avatarNormal,
-
-            // Background color isn't added for room avatar because it changes it's shape to a square
-            this.props.isChatRoom ? {} : {backgroundColor: themeColors.icon},
+            this.props.size === CONST.AVATAR_SIZE.SMALL ? styles.avatarSmall : styles.avatarNormal,
             ...this.props.imageStyles,
         ];
 
+        const AVATAR_SIZES = {
+            [CONST.AVATAR_SIZE.DEFAULT]: variables.avatarSizeNormal,
+            [CONST.AVATAR_SIZE.SUBSCRIPT]: variables.avatarSizeSubscript,
+            [CONST.AVATAR_SIZE.SMALL]: variables.avatarSizeSmall,
+            [CONST.AVATAR_SIZE.LARGE]: variables.avatarSizeLarge,
+        };
+        const iconSize = AVATAR_SIZES[this.props.size];
```

Comment:
NAB, but this logic feels like it could be in [`StyleUtils`](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) as a `getAvatarSize()` or something